dlib/python/numpy_image.h

```diff
@@ -70,11 +73,9 @@ class numpy_rgb_image
     numpy_rgb_image (boost::python::object& img) 
     {
         long shape[3];
-        get_numpy_ndarray_parts(img, _data, shape);
+        get_numpy_ndarray_parts(img, _data, _contig_buf, shape);
         _nr = shape[0];
         _nc = shape[1];
-        if (shape[2] != 3)
```

Comment:
Yeah. A lot of thought was given to it. Since the ndarray is very likely to be non-contiguous (e.g. ROI of an image), copying the data twice is obviously unwise. I will rethink about it.